test_files/quote_props/quote.ts

```diff
@@ -0,0 +1,23 @@
+interface Quoted {
+  [k: string]: number;
+}
+let quoted: Quoted = {};
+
+console.log(quoted.hello);
+quoted.hello = 1;
+quoted['hello'] = 1;
+
+interface QuotedMixed extends Quoted {
+  // Assume that foo should be renamed, as it is explicitly declared.
+  // It's unclear whether it's the right thing to do, user code might
```

Comment:
in the past we've thought that tsickle should produce a diagnostic for this case, which would be treated as a build error in tsc-wrapped.
You could still get around that with
`type QuotedMixed = Quoted & {foo: number};`
because type intersections are totally unchecked. Maybe that's good since it provides a way to suppress the tsickle check.